cpp/ql/src/Security/CWE/CWE-190/TaintedAllocationSize.ql

```diff
@@ -42,6 +42,12 @@ class TaintedAllocationSizeConfiguration extends TaintTrackingConfiguration {
       e instanceof AssignArithmeticOperation
     ) and
     not convertedExprMightOverflow(e)
+    or
+    // Subtracting two pointers is either well-defined (and the result will likely be small), or
+    // terribly undefined and dangerous. Here, we assume that the programmer has ensured that the
+    // result is well-defined (i.e., the two pointers point to the same object), and thus the result
+    // will likely be small.
```

Comment:
Seems reasonable.  If the subtraction result is not well-defined, that should probably be handled by another query (which may or may-not already exist) rather than blaming a later `malloc`.

Reply:
Indeed. 👍 I've opened up https://github.com/github/codeql-c-team/issues/525 for this exact query idea.